core/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -132,7 +133,11 @@ public static BulkItemResponse fromXContent(XContentParser parser, int id) throw
                 if (token == XContentParser.Token.START_OBJECT) {
                     exception = ElasticsearchException.fromXContent(parser);
                 }
-            } else if (STATUS.equals(currentFieldName) == false) {
+            } else if (STATUS.equals(currentFieldName)) {
+                if (token == XContentParser.Token.VALUE_NUMBER) {
+                    status = RestStatus.fromCode(parser.intValue());
+                }
```

Comment:
what to do here otherwise? not really likely to happen that it's not a number I guess... maybe leniency is good here